### PR TITLE
Remove dedicated SSL network write buffer

### PR DESCRIPTION
**Pull request by Tim-Brooks**
This is related to #27260. Currently for the SSLDriver we allocate a
dedicated network write buffer and encrypt the data into that buffer one
buffer at a time. This requires constantly switching between encrypting
and flushing. This commit adds a dedicated outbound buffer for SSL
operations that will internally allocate new packet sized buffers as
they are need (for writing encrypted data). This allows us to totally
encrypt an operation before writing it to the network. Eventually it can
be hooked up to buffer recycling.

### CODE REVIEW

**Comment by s1monw** [comment]
I don't feel I am comfortable enough with SSL code to give a good review. I looked over it and nothing looks terrible. So I am relying on @original-brownbear for this.

**Comment by original-brownbear** [comment]
I'm on it today, sorry for missing this one!

File: libs/nio/src/main/java/org/elasticsearch/nio/Page.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+package org.elasticsearch.nio;
+
+import org.elasticsearch.common.util.concurrent.AbstractRefCounted;
+
+import java.io.Closeable;
+import java.nio.ByteBuffer;
+
+public class Page implements Closeable {
+
+    private static final Runnable DEFAULT_CLOSEABLE = () -> {};
```
Comment (original-brownbear) [inline libs/nio/src/main/java/org/elasticsearch/nio/Page.java]: NIT: why not just inline this lambda?

**Review by original-brownbear — APPROVED** [review]
LGTM :)

**Comment by tbrooks8** [comment]
@original-brownbear Thanks. I've made your suggested changes and will merge once CI passes.

**Pull request merged**